src/plugins/expressions/common/execution/execution.ts

```diff
@@ -104,10 +107,10 @@ export class Execution<
   private readonly abortRejection = abortSignalToPromise(this.abortController.signal);
 
   /**
-   * Races a given promise against the "abort" event of `abortController`.
+   * Races a given observable against the "abort" event of `abortController`.
    */
-  private race<T>(promise: Promise<T>): Promise<T> {
-    return Promise.race<T>([this.abortRejection.promise, promise]);
+  private race<T>(observable: Observable<T>): Observable<T> {
+    return race(from(this.abortRejection.promise), observable);
```

Comment:
I think you can get rid of `abortRejection` and the `race` function, and replace them with `takeUntil`, where `race` was previously used (Like we do in `x-pack/plugins/data_enhanced/common/search/poll_search.ts`)

```
observable.pipe(
   takeUntil(fromEvent(this.abortController.signal, 'abort'))
);
```

Reply:
Then it won't produce a value, and therefore it won't call a subscriber. The promise, in that case, will be resolved with `undefined`, which is not what we expect. I guess the correct solution here is to stick to the race operator to keep existing behavior and call our subscription as soon as it's canceled.

Reply:
The behavior difference is that if the current version of the aborted promise is already fulfilled, then it will emit straightaway. 
@lizozom, It think this wouldn't work the same in your snippet, but I am not sure if this is relevant here.